Use C++ filesystem library/iostreams, added unit tests (#911)

uweseimet opened the pull request:
Summary of changes:

- Use C++ filesystem library and C++ iostreams for almost all file operations (gcc < 10.3.0 from buster prevents a complete replacement)
- Removed obsolete Filepath class
- Removed duplicate code
- Renamed fileio.cpp/h to rasdump_fileio.cpp/h, because only rasump still uses this class
- Fixed more SonarQube issues
- Added unit tests, total test count is 264

akuker reviewed (approved):
Approved! Merge away!

uweseimet commented:
@akuker Please note that I applied two minor changes after your approval and before merging, because the merge result was causing some issues. I hope (and assume) you don't mind these changes:

1. clang++ still did not compile in my environment because of the LUN_MAX issue I mentioned in your PR. I fixed that.
2. I added an #ifdef in rascsi.cpp to prevent "SBC_Version::Init();" to be compiled on a standard Linux PC. We have one or two similar #ifdefs for that purpose elsewhere. For running/testing anything where you do not need the RaSCSI hardware for (that's the majority of the code) it is important that you can launch rascsi and execute remote commands on a regular PC, where an IDE like Eclipse is available.

In my environment everything compiles with clang++ without any issues now.
